Libraries/IonAuth.php

```diff
@@ -0,0 +1,440 @@
+
+		$emailActivation = $this->config->emailActivation;
+
+		$id = $this->ionAuthModel->register($identity, $password, $email, $additional_data, $group_ids);
+
+		if (! $emailActivation)
+		{
+			if ($id !== false)
+			{
+				$this->setMessage('IonAuth.account_creation_successful');
+				$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_successful']);
+				return $id;
+			}
+			else
+			{
+				$this->setError('IonAuth.account_creation_unsuccessful');
+				$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_unsuccessful']);
+				return false;
+			}
+		}
+		else
+		{
+			if (! $id)
+			{
+				$this->setError('IonAuth.account_creation_unsuccessful');
+				return false;
+			}
+
+			// deactivate so the user must follow the activation flow
+			$deactivate = $this->ionAuthModel->deactivate($id);
+
+			// the deactivate method call adds a message, here we need to clear that
+			$this->ionAuthModel->clearMessages();
+
+			if (! $deactivate)
+			{
+				$this->setError('IonAuth.deactivate_unsuccessful');
+				$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_unsuccessful']);
+				return false;
+			}
+
+			$activationCode = $this->ionAuthModel->activation_code;
+			$identity       = $this->config->identity;
+			$user           = $this->ionAuthModel->user($id)->row();
+
+			$data = [
+				'identity'   => $user->{$identity},
+				'id'         => $user->id,
+				'email'      => $email,
+				'activation' => $activationCode,
+			];
+			if (! $this->config->useCiEmail)
+			{
+				$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_successful', 'activation_email_successful']);
+				$this->setMessage('IonAuth.activation_email_successful');
+				return $data;
+			}
+			else
+			{
+				$message = $this->load->view($this->config->emailTemplates . $this->config->emailActivate, $data, true);
+
+				$this->email->clear();
+				$this->email->from($this->config->adminEmail, $this->config->siteTitle);
+				$this->email->to($email);
+				$this->email->subject($this->config->siteTitle . ' - ' . $this->lang->line('emailActivation_subject'));
+				$this->email->message($message);
+
+				if ($this->email->send() === true)
+				{
+					$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_successful', 'activation_email_successful']);
+					$this->setMessage('IonAuth.activation_email_successful');
+					return $id;
+				}
+			}
+
+			$this->ionAuthModel->triggerEvents(['post_account_creation', 'post_account_creation_unsuccessful', 'activation_email_unsuccessful']);
+			$this->setError('IonAuth.activation_email_unsuccessful');
+			return false;
+		}
+	}
+
+	/**
+	 * Logout
+	 *
+	 * @return true
+	 * @author Mathew
+	 */
+	public function logout(): bool
+	{
+		$this->ionAuthModel->triggerEvents('logout');
+
+		$identity = $this->config->identity;
+
+		session()->remove([$identity, 'id', 'user_id']);
```

Comment:
use this->session instead since it's available
```suggestion
		$this->session->remove([$identity, 'id', 'user_id']);
```